flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -169,6 +169,9 @@ BufferAndBacklog pollBuffer() {
 				BufferConsumer bufferConsumer = buffers.peek();
 
 				buffer = bufferConsumer.build();
+				if (!isLocal && bufferCompressor != null && buffer.isBuffer() && buffer.readableBytes() > 0) {
```

Comment:
`bufferCompressor != null && buffer.isBuffer() && buffer.readableBytes() > 0` can be replaced by above `canBeCompressed(buffer)`